File: C(GCC5.4.1)/ABC/ABC081/ABC081_A_v01.c
// contest  : ABC 081
// task     : A
// URL      : https://atcoder.jp/contests/abc081/tasks/abc081_a
// compiler : C (GCC 5.4.1)
// status   : https://atcoder.jp/contests/abc081/submissions/9171731 : AC

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

#define num_cells 3

void ABC081A (void)
{
	// local variables for this function
	char         given_data[num_cells];
	int          retval_scanf;
	unsigned int num_marbles;

	// STEP.01
	// read out the given string
	retval_scanf = scanf("%s", given_data);

	// STEP.02
	// find the number of squares on which Sunuke will place a marble
	num_marbles = (unsigned int)(0);

	for (int iter = 0; iter < num_cells; ++iter)
	{
		if ( given_data[iter] == '1' ) ++num_marbles;
	}

	// STEP.03
	// print the answer of this task to the standard output
	printf( "%d\n", num_marbles );

	// STEP.END
	return;
}

int main (void)
{
	// STEP.01
	// calculate & output the answer
	ABC081A();

	// STEP.END
	return EXIT_SUCCESS;
}

/* EOF */

File: Rust(1.15.1)/ABC/ABC081/ABC081_A_v01.rs
// contest  : ABC 081
// task     : A
// URL      : https://atcoder.jp/contests/abc081/tasks/abc081_a
//          : https://atcoder.jp/contests/abs/tasks/abc081_a
// compiler : Rust (1.15.1)
// status   : https://atcoder.jp/contests/abc081/submissions/9196593 : AC

// * reference
//   * https://qiita.com/tubo28/items/e6076e9040da57368845

use std::io::*;
use std::str;

fn main () {
    let test_data: u8 = read();

    println!( "{}", abc081_a(test_data) );
}

fn abc081_a (test_data: u8) -> u8 {

    let mut iter:        u8 = test_data;
    let mut num_marbles: u8 = 0;

    while iter > 0 {
        num_marbles += iter % 10;
        iter        /= 10;
    }

    /* return value of this function */
    num_marbles

}

// * reference
fn read<T: std::str::FromStr>() -> T {

    let stdin = std::io::stdin();

    let handle = stdin.lock();

    let token: String = handle
        .bytes()
        .map(|c| c.expect("failed to read char") as char)
        .skip_while(|c|  c.is_whitespace())
        .take_while(|c| !c.is_whitespace())
        .collect();

    token.parse().ok().expect("failed to parse token")

}

/* EOF */
